src/toil/wdl/wdltoil.py

```diff
@@ -911,14 +994,24 @@ def devirtualize_to(filename: str, dest_dir: str, file_source: Union[AbstractFil
                 virtualized_to_devirtualized[filename] = result
             logger.debug('Devirtualized %s as openable file %s', filename, result)
         else:
-            # This is a local file
-            # To support relative paths, join the execution dir and filename
-            # if filename is already an abs path, join() will do nothing
-            if execution_dir is not None:
-                result = os.path.join(execution_dir, filename)
-            else:
-                result = filename
-            logger.debug("Virtualized file %s is already a local path", filename)
+            result = None
+            if wdl_options.get("filename_to_uri") is not None:
```

Comment:
I think Files coerced from strings will still see these imported paths from the JSON URL. I'm currently not too sure how to avoid that with this implementation; I went with this implementation originally as I *think* this is the simplest solution overall that avoids "hacking" around miniwdl's internals (such as abusing setattr).

Reply:
I don't understand what it means for a File instance to "see" an imported path that came from the input JSON file's URL.

Reply:
I changed the implementation to use setattr so the above comment doesn't really apply anymore. A file isn't going to "see" an imported path anymore as I poll for the remote paths at startup.